src/app/shared/wysiwyg/wysiwyg.component.ts

```diff
@@ -1,39 +1,40 @@
-import { Component, Input, OnInit, OnChanges, SimpleChanges, OnDestroy, ElementRef, ViewChild } from '@angular/core';
+import { Component, Input, OnInit, OnChanges, SimpleChanges, OnDestroy, ElementRef, ViewChild, ChangeDetectorRef } from '@angular/core';
 import { NgModel } from '@angular/forms';
 import { BaseModel } from '../model/base.model';
 import { ImageModel } from '../model/image.model';
 import { ProseMirrorMarkdownEditor, ProseMirrorImage } from './prosemirror.markdown.editor';
-import { PromptComponent } from './prompt.component';
 
 @Component({
   selector: 'publish-wysiwyg',
   template: `
     <div #contentEditable [class.changed]="changed" [class.invalid]="invalid"></div>
     <p *ngIf="invalid" class="error">{{invalid | capitalize}}</p>
     
-    <publish-prompt #prompt *ngIf="!editor?.isSelectionEmpty()">
```

Comment:
I made a terrible error in the way I managed visible state in the prompt component. I was calling show and hide from the #prompt template variable in this parent component. This bypasses change detection.

I could have passed visible in as an input to prompt component, but it stopped making sense to keep it as a separate component. The only benefit seems like a separate place to keep modal CSS, but it was using deep selectors, so... not ideal.

So basically, prompt.component can die in a fire (and I can learn to use Angular properly.)